### PR TITLE
Add workflow to add/update offline docs

### PR DESCRIPTION
**Pull request by L1ghtmann**
What does this implement/fix? Explain your changes.
---------------------------------------------------
See title

Does this close any currently open issues?
------------------------------------------
No, but it will tick another item off of #308

Any relevant logs, error output, etc?
-------------------------------------

Any other comments?
-------------------
Tested on [separate branch](https://github.com/L1ghtmann/theos/tree/my-docs) and worked like a charm. Would be nice if we could only run the workflow if theos/theos.dev is updated, but not sure that's possible. 

~~Edit: CSS looks to be borked atm~~

Where has this been tested?
---------------------------
**Operating System:** …

Linux (WSL)

**Platform:** …

**Target Platform:** …

**Toolchain Version:** …

**SDK Version:** …

### CODE REVIEW

File: .github/workflows/offline-docs.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        run: |
+          sed -i '/<svg.*12.*/,/<\/svg>/d' docs/*.html
+          sed -i 's/ Deployed on <a href="https:\/\/netlify.com">Netlify<\/a>.//g' docs/*.html
+
+      - name: Update docs
+        run: |
+          git config --global user.name "$GITHUB_ACTOR"
+          git config --global user.email "$GITHUB_ACTOR@users.noreply.github.com"
+          git add docs
+          git commit --all -m "[docs] Update offline docs to the latest commit" && git push || echo "No changes were made"
```
Comment (leptos-null) [inline .github/workflows/offline-docs.yml:40]: On the line above, you do an explicit add. I think there should not be a `--all` here.

```suggestion
          git commit -m "[docs] Update offline docs to the latest commit" && git push || echo "No changes were made"
```

File: .github/workflows/offline-docs.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7.2'
+
+      - name: Build docs
+        run: |
+          git clone --depth=1 --recursive https://github.com/theos/theos.dev tmp
```
Comment (leptos-null) [inline .github/workflows/offline-docs.yml:24]: nit: use `actions/checkout` for best GitHub integration

```yml
uses: actions/checkout@v3
with:
  repository: theos/theos.dev
  path: tmp
  submodules: recursive

File: .github/workflows/offline-docs.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          cd tmp
+          make all
```
Comment (leptos-null) [inline .github/workflows/offline-docs.yml:26]: consolidate with `make -C tmp`

also fixes what appears to be a bug below: the current working directory is `tmp` and then we `rm -rf tmp`

Comment (L1ghtmann) [reply .github/workflows/offline-docs.yml]: Thanks for catching all of these!

**Review by leptos-null — COMMENTED** [review]
Overall this code looks good. The workflow will not work though, for two reasons:

1. `secrets.GITHUB_TOKEN` does not have sufficient permissions to write to this repository/ branch, I believe
2. the trigger for this workflow is `on: push: branches: master` however this is unrelated to the event that we're interested in, which is pushes to `master` on `theos/theos.dev`
    - both Lightmann and I originally thought this would work because `update-submodules` would run on organization pushes, however pushes to `theos.dev` would not result in a commit to this repository and therefore would not trigger this workflow

**Review by uroboro — APPROVED** [review]

**Comment by leptos-null** [comment]
The first part of my previous comment was not correct: `secrets.GITHUB_TOKEN` does have access to write to this repository

File: .github/workflows/offline-docs.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
+name: Update Offline Docs
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'master'
```
Comment (leptos-null) [inline .github/workflows/offline-docs.yml:7]: In order to respond to pushes on `theos.dev`, we would need a repository dispatch `on` trigger.

We can re-use the organization-wide trigger for this. I think that ideally `theos.dev` would have it's own trigger, but I don't think it's a big enough deal to setup a separate handler for.

```suggestion
  repository_dispatch:
    types: [org_push]
```

**Review by leptos-null — APPROVED** [review]
Looks good to me.

Thanks for checking a big item off the to-do list.

**Pull request merged**